caas/kubernetes/provider/specs/legacy_test.go

```diff
@@ -0,0 +1,451 @@
+			},
+			Config: map[string]interface{}{
+				"foo":        "bar",
+				"restricted": "'yes'",
+				"switch":     true,
+			},
+			ProviderContainer: &k8sspecs.K8sContainerSpec{
+				ImagePullPolicy: "Always",
+			},
+		}}
+
+		pSpecs.ProviderPod = &k8sspecs.K8sPodSpec{
+			ActiveDeadlineSeconds:         int64Ptr(10),
+			RestartPolicy:                 core.RestartPolicyOnFailure,
+			TerminationGracePeriodSeconds: int64Ptr(20),
+			SecurityContext: &core.PodSecurityContext{
+				RunAsNonRoot:       boolPtr(true),
+				SupplementalGroups: []int64{1, 2},
+			},
+			Hostname:          "host",
+			Subdomain:         "sub",
+			PriorityClassName: "top",
+			Priority:          int32Ptr(30),
+			DNSConfig: &core.PodDNSConfig{
+				Nameservers: []string{"ns1", "ns2"},
+			},
+			DNSPolicy: "ClusterFirstWithHostNet",
+			ReadinessGates: []core.PodReadinessGate{
+				{ConditionType: core.PodScheduled},
+			},
+			Service: &k8sspecs.K8sServiceSpec{
+				Annotations: map[string]string{"foo": "bar"},
+			},
+			KubernetesResources: &k8sspecs.KubernetesResources{
+				ServiceAccount: &k8sspecs.ServiceAccountSpec{
+					Name:                         "serviceAccountFoo",
+					AutomountServiceAccountToken: boolPtr(true),
+				},
+				CustomResourceDefinitions: map[string]apiextensionsv1beta1.CustomResourceDefinitionSpec{
+					"tfjobs.kubeflow.org": {
+						Group:   "kubeflow.org",
+						Version: "v1alpha2",
+						Scope:   "Namespaced",
+						Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+							Kind:     "TFJob",
+							Plural:   "tfjobs",
+							Singular: "tfjob",
+						},
+						Validation: &apiextensionsv1beta1.CustomResourceValidation{
+							OpenAPIV3Schema: &apiextensionsv1beta1.JSONSchemaProps{
+								Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+									"tfReplicaSpecs": {
+										Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+											"PS": {
+												Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+													"replicas": {
+														Type: "integer", Minimum: float64Ptr(1),
+													},
+												},
+											},
+											"Chief": {
+												Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+													"replicas": {
+														Type:    "integer",
+														Minimum: float64Ptr(1),
+														Maximum: float64Ptr(1),
+													},
+												},
+											},
+											"Worker": {
+												Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+													"replicas": {
+														Type:    "integer",
+														Minimum: float64Ptr(1),
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		return pSpecs
+	}
+
+	spec, err := k8sspecs.ParsePodSpec(specStrBase)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(spec, jc.DeepEquals, getExpectedPodSpecBase())
+}
+
+func float64Ptr(f float64) *float64 {
```

Comment:
there are defined in a couple of places, can we consolidate?

Reply:
added a TODO to move these to testing package.